cpp/test/sg/umap_parametrizable_test.cu

```diff
@@ -0,0 +1,306 @@
+    }
+
+    if (test_params.knn_params) {
+      delete knn_indices_b;
+      delete knn_dists_b;
+    }
+  }
+
+  void assertions(cumlHandle& handle, float* X, float* embedding_ptr,
+                  TestParams& test_params, UMAPParams& umap_params) {
+    cudaStream_t stream = handle.getStream();
+    auto alloc = handle.getDeviceAllocator();
+    int& n_samples = test_params.n_samples;
+    int& n_features = test_params.n_features;
+
+    ASSERT_TRUE(!has_nan(embedding_ptr, n_samples * umap_params.n_components,
+                         alloc, stream));
+
+    double trustworthiness = trustworthiness_score<float, EucUnexpandedL2Sqrt>(
+      handle, X, embedding_ptr, n_samples, n_features, umap_params.n_components,
+      umap_params.n_neighbors);
+
+    std::cout << "min. expected trustworthiness: "
+              << test_params.min_trustworthiness << std::endl;
+    std::cout << "trustworthiness: " << trustworthiness << std::endl;
+    ASSERT_TRUE(trustworthiness > test_params.min_trustworthiness);
+  }
+
+  void test(TestParams& test_params, UMAPParams& umap_params) {
+    std::cout << "\numap_params : [" << std::boolalpha
+              << umap_params.n_neighbors << "-" << umap_params.n_components
+              << "-" << umap_params.n_epochs << "-" << umap_params.random_state
+              << "-" << umap_params.multicore_implem << "]" << std::endl;
+
+    std::cout << "test_params : [" << std::boolalpha
+              << test_params.fit_transform << "-" << test_params.supervised
+              << "-" << test_params.knn_params << "-" << test_params.n_samples
+              << "-" << test_params.n_features << "-" << test_params.n_clusters
+              << "-" << test_params.min_trustworthiness << "]" << std::endl;
+
+    cumlHandle handle;
+    cudaStream_t stream = handle.getStream();
+    auto alloc = handle.getDeviceAllocator();
+    int& n_samples = test_params.n_samples;
+    int& n_features = test_params.n_features;
+
+    UMAPAlgo::find_ab(&umap_params, alloc, stream);
+
+    device_buffer<float> X_d(alloc, stream, n_samples * n_features);
+    device_buffer<int> y_d(alloc, stream, n_samples);
+
+    Random::make_blobs<float, int>(X_d.data(), y_d.data(), n_samples,
+                                   n_features, test_params.n_clusters, alloc,
+                                   stream, nullptr, nullptr);
+
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+
+    MLCommon::LinAlg::convert_array((float*)y_d.data(), y_d.data(), n_samples,
+                                    stream);
+
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+
+    device_buffer<float> embeddings1(alloc, stream,
+                                     n_samples * umap_params.n_components);
+    float* e1 = embeddings1.data();
+
+    get_embedding(handle, X_d.data(), (float*)y_d.data(), e1, test_params,
+                  umap_params);
+
+    assertions(handle, X_d.data(), e1, test_params, umap_params);
+
+    /*
+    if (!umap_params.multicore_implem) {
```

Comment:
Remove commented code.

Reply:
Actually, this is part of the code should have been re-enabled.